File: matrix_multi/matrix.java
class Matrix {
  public static void main(String args[])
  {
    if (args.length != 1) {
      System.out.println("usage: java Matrix N");
      return;
    }
    int maxN = Integer.parseInt(args[0]);
    int n;
    for(n = 0; n <= maxN; n++){
      double[][] a = new double[n][n]; // Matrix A
      double[][] b = new double[n][n]; // Matrix B
      double[][] c = new double[n][n]; // Matrix C

    // Initialize the matrices to some values.
      int i, j;
      for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++) {
          a[i][j] = i * n + j;
          b[i][j] = j * n + i;
          c[i][j] = 0;
        }
      }

      long begin = System.currentTimeMillis();
      int k;

    /**************************************/
    /* Write code to calculate C = A * B. */
      for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++) {
          for (k = 0; k < n; k++){
            c[i][j]=c[i][j]+a[i][k]*b[k][j];
          }
        }
      }
      long end = System.currentTimeMillis();
      System.out.printf("%d %.6f\n", n,(end - begin)/1000.0);
    }
  }
}

File: matrix_multi/matrix.py
import matplotlib.pyplot as plt
import numpy, sys, time
x = []
y = []
######################################
# Write code to calculate C = A * B. #
for n in xrange(300):
    a = numpy.zeros((n, n)) # Matrix A
    b = numpy.zeros((n, n)) # Matrix B
    c = numpy.zeros((n, n)) # Matrix C

# Initialize the matrices to some values.
    for i in xrange(n):
        for j in xrange(n):
            a[i, j] = i * n + j
            b[i, j] = j * n + i
            c[i, j] = 0
    begin = time.time()
    for i in xrange(n):
        for j in xrange(n):
            result = 0
            for k in xrange(n):
                result = result + a[i,k]*b[k,j]
            c[i,j]= result
    end = time.time()
    x.append(n)
    y.append(end-begin)

plt.plot(x,y)
plt.title("python matrix")
plt.xlabel("matrix number")
plt.ylabel("execution time(s)")
plt.savefig("python_matrix.png")
